### ISSUE TITLE
Better API for bots?

### ISSUE THREAD

**Issue by overlisted** [benawad/dogehouse#1207]
It would be really nice if the current Elixir API supported a "bot mode" and "bot-only" accounts, which would:
- Be able to **join multiple rooms at once**
- **Have a special badge** after their display name showing that the user is a bot
- Be able to implement do-it-yourself sharding, which would need to **log into the account with multiple clients at once**
- **Get rate limited by the API** if they are performing too many actions on one room

**Comment by nadirabbas** [comment]
yeah, rate limiting is really importing, bots are messing up the experience

**Comment by drvladb** [comment]
> bots are messing up the experience  

A temporary workaround is to write to a queue that is dispatched every second.

**Comment by benawad** [comment]
> yeah, rate limiting is really important, bots are messing up the experience

what are they doing right now?

**Comment by nadirabbas** [comment]
They join, send a msg, leave, join, send a message, leave. Doing it really fast.

**Comment by overlisted** [comment]
Also, room admins probably should be able to turn off bots or allow only some bots to join

**Comment by TheOtterlord** [comment]
> Also, room admins probably should be able to turn off bots or allow only some bots to join

I think using an allow-list like discord (& I think twitch) would work the best. Then admins have full control over the bots

**Comment by jamesql** [comment]
Is anybody working on this? if not is anyone willing to help me out on the front end part sometime?

**Comment by HonbraDev** [comment]
I own a bot. My current workaround is a queue that sends a message every second.

**Comment by JackCrispy** [comment]
> Also, room admins probably should be able to turn off bots or allow only some bots to join

Yeah which is more reason to have application authentication like discord.com/developers does it. Having an ID for each bot and token for each bot which doesn’t change.

Also is it even possible with the current architecture to allow accounts to join multiple rooms?

**Comment by dragonismcode** [comment]
Adding onto this, having a regenerable token. If you're bot's token gets leaked you're screwed.

_cough cough_ Honbra leaking my bot account's token _cough cough_

**Comment by jamesql** [comment]
@dragonismcode I think deleting accounts still delete everything so you should be able to regen token that way for now
